telemetry_api/lib/telemetry_api/traces.ex

```diff
@@ -0,0 +1,189 @@
+defmodule TelemetryApi.Traces do
+  @moduledoc """
+  The Traces context.
+  """
+  alias TelemetryApi.Traces.Trace
+  alias TelemetryApi.Operators
+  alias TelemetryApi.ContractManagers.StakeRegistry
+
+  require OpenTelemetry.Tracer
+  require OpenTelemetry.Ctx
+  alias OpenTelemetry.Tracer, as: Tracer
+  alias OpenTelemetry.Ctx, as: Ctx
+
+  @doc """
+  Send the trace to OpenTelemetry
+
+  This function is responsible for creating a new span and storing the context in the Agent.
+
+  ## Examples
+
+      iex> merkle_root = "0x1234567890abcdef"
+      iex> create_task_trace(merkle_root)
+      :ok
+  """
+  def create_task_trace(merkle_root) do
+    # Get total stake
+    with {:ok, total_stake} <- StakeRegistry.get_current_total_stake() do
+      span_ctx =
+        Tracer.start_span(
+          "Task: #{merkle_root}",
+          %{
+            attributes: [
+              {:merkle_root, merkle_root},
+              {:total_stake, total_stake}
+            ]
+          }
+        )
+
+      ctx = Ctx.get_current()
+
+      TraceStore.store_trace(merkle_root, %Trace{
+        parent_span: span_ctx,
+        context: ctx,
+        total_stake: total_stake,
+        current_stake: 0,
+        responses: []
+      })
+
+      IO.inspect("New task trace with merkle_root: #{IO.inspect(merkle_root)}")
+      :ok
+    end
+  end
+
+  @doc """
+  Registers an operator response in the task trace.
+
+  ## Examples
+
+      iex> merkle_root = "0x1234567890abcdef"
+      iex> operator_id = "0x..."
+      iex> register_operator_response(merkle_root, operator_id)
+      :ok
+  """
+  def register_operator_response(merkle_root, operator_id) do
+    with {:ok, operator} <- Operators.get_operator_by_id(operator_id),
+         {:ok, trace} <- set_current_trace(merkle_root) do
+      operator_stake = String.to_integer(operator.stake)
+      new_stake = trace.current_stake + operator_stake
+      new_stake_fraction = new_stake / trace.total_stake
+      operator_stake_fraction = operator_stake / trace.total_stake
```

Comment:
Are these numbers not `Decimal`? I think they should be, ints in postgres are not big enough to handle some of these numbers.
Then, i.e. for divisioind, the following is the way to divide Decimals: `Decimal.div(operator_stake, total_stake)`

Reply:
We are storing the stakes as strings in the db. I have changed them to `Decimals` in Elixir now.